flash/vision/detection/model.py

```diff
@@ -11,19 +11,33 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any, Callable, Mapping, Sequence, Type, Union
+from typing import Any, Callable, Mapping, Optional, Sequence, Type, Union
 
 import torch
 import torchvision
 from torch import nn
 from torch.optim import Optimizer
+from torchvision.ops import box_iou
 
-from flash.core.classification import ClassificationTask
+from flash.core import Task
+from flash.vision.detection.data import _default_transform
+from flash.vision.embedding.image_embedder_model import ImageEmbedderDataPipeline
 
 _models = {"fasterrcnn_resnet50_fpn": torchvision.models.detection.fasterrcnn_resnet50_fpn}
 
 
-class ImageDetector(ClassificationTask):
+def _evaluate_iou(target, pred):
+    """
+    Evaluate intersection over union (IOU) for target from dataset and output prediction
+    from model
+    """
+    if pred["boxes"].shape[0] == 0:
+        # no box detected, 0 IOU
+        return torch.tensor(0.0, device=pred["boxes"].device)
+    return box_iou(target["boxes"], pred["boxes"]).diag().mean()
```

Comment:
I don't know about the `.diag().mean()` part. What does say? Should we try to implement f1_score based on precision and recall?

From the repo readme:
>It also has the best practices in AI research embedded into each task so you don't have to be a deep learning PhD to leverage its power :)

Reply:
Hi @adriantre, you could check out detection metrics [here](https://github.com/rafaelpadilla/Object-Detection-Metrics) and box_iou [here](https://pytorch.org/docs/stable/_modules/torchvision/ops/boxes.html#box_iou). Yes, I will the task more clear and refactor it. It's Work in Progress.

Reply:
Sounds good ! But let's merge it with Lightning Metrics :)

Reply:
I see this PR is merged now. Should we create an issue on metrics so we dont forget to
> embed best practices in AI research into it
as Flash promises?

Reply:
Hey @adriantre, yes, I have it in my mind. I wanted to merge it with the Lightning Metrics. But Waiting for [mAP](https://github.com/PyTorchLightning/pytorch-lightning/pull/4564) metric to get merged. We could use it.

Reply:
Ref: https://cocodataset.org/#detection-eval It's a standard metric to evaluate Detection on IOU threshold.

Reply:
Sound good, great job!